framework/src/meshgenerators/Poly2TriMeshGenerator.C

```diff
@@ -0,0 +1,242 @@
+  params.addParam<MooseEnum>(
+      "algorithm",
+      algorithm,
+      "Control the use of binary search for the nodes of the stitched surfaces.");
+  params.addParam<bool>(
+      "verbose_stitching", false, "Whether mesh stitching should have verbose output.");
+
+  params.addClassDescription("Triangulates meshes within boundaries defined by input meshes.");
+
+  return params;
+}
+
+Poly2TriMeshGenerator::Poly2TriMeshGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _bdy_ptr(getMesh("boundary")),
+    _interpolate_bdy(getParam<unsigned int>("interpolate_boundary")),
+    _refine_bdy(getParam<bool>("refine_boundary")),
+    _smooth_tri(getParam<bool>("smooth_triangulation")),
+    _hole_ptrs(getMeshes("holes")),
+    _stitch_holes(getParam<std::vector<bool>>("stitch_holes")),
+    _interpolate_holes(getParam<std::vector<unsigned int>>("interpolate_holes")),
+    _refine_holes(getParam<std::vector<bool>>("refine_holes")),
+    _desired_area(getParam<Real>("desired_area")),
+    _desired_area_func(getParam<std::string>("desired_area_func")),
+    _algorithm(parameters.get<MooseEnum>("algorithm")),
+    _verbose_stitching(parameters.get<bool>("verbose_stitching"))
+{
+  if (!_stitch_holes.empty() && _stitch_holes.size() != _hole_ptrs.size())
+    paramError("stitch_holes", "Need one stitch_holes entry per hole, if specified.");
+
+  if (!_interpolate_holes.empty() && _interpolate_holes.size() != _hole_ptrs.size())
+    paramError("interpolate_holes", "Need one interpolate_holes entry per hole, if specified.");
+}
+
+std::unique_ptr<MeshBase>
+Poly2TriMeshGenerator::generate()
+{
+  // We put the boundary mesh in a local pointer
+  std::unique_ptr<UnstructuredMesh> mesh = dynamic_pointer_cast<UnstructuredMesh>(_bdy_ptr);
+
+  Poly2TriTriangulator poly2tri(*mesh);
+  poly2tri.triangulation_type() = TriangulatorInterface::PSLG;
+
+  poly2tri.set_interpolate_boundary_points(_interpolate_bdy);
+  poly2tri.set_refine_boundary_allowed(_refine_bdy);
+
+  poly2tri.desired_area() = _desired_area;
+  poly2tri.minimum_angle() = 0; // Not yet supported
+  poly2tri.smooth_after_generating() = _smooth_tri;
+
+  if (_desired_area_func != "")
+  {
+    // poly2tri will clone this so it's fine going out of scope
+    ParsedFunction<Real> area_func{_desired_area_func};
+    poly2tri.set_desired_area_function(&area_func);
+  }
+
+  std::vector<TriangulatorInterface::MeshedHole> meshed_holes;
+  std::vector<TriangulatorInterface::Hole *> triangulator_hole_ptrs;
+
+  // Make sure pointers here aren't invalidated by a resize
+  meshed_holes.reserve(_hole_ptrs.size());
+  for (auto hole_i : index_range(_hole_ptrs))
+  {
+    meshed_holes.emplace_back(*_hole_ptrs[hole_i]->get());
+    if (hole_i < _refine_holes.size())
+      meshed_holes.back().set_refine_boundary_allowed(_refine_holes[hole_i]);
+    if (_interpolate_holes.size() > hole_i)
+      libmesh_not_implemented();
```

Comment:
so "interpolation" or having more than one node per segment in holes is not implemented?

Reply:
Yeah; it turned out to be less trivial and more redundant (it's easy enough to refine the input defining the hole) than I anticipated.  I'll remove the parameter.